### PR TITLE
Add a new control AspectRatioLayout

### PR DESCRIPTION
**Pull request by WCKYWCKF**
The control AspectRatioLayout provides a feature to select control layout based on the aspect ratio of the visual area (Bounds).

### CODE REVIEW

**Comment by WCKYWCKF** [comment]
Providing a default value in AspectRatioLayout.ItemsProperty causes all AspectRatioLayouts to share a List. This issue has been fixed, please help review it.

**Comment by rabbitism** [comment]
My suggestions:
the `AspectRatioMode` concept kinda limit extensibility. I would recommend introduce something like 
```csharp
public AvaloniaList<double> RatioBreakpoints {get; set; }
```
for Layout,
and 
```csharp
public double MinRatio { get; set; }
public double MaxRaio { get; set; }
```
for layout item. 

when bounds ratio change, find the first item that fall into it's range.
Any thoughts?

**Comment by WCKYWCKF** [comment]
> My suggestions: the `AspectRatioMode` concept kinda limit extensibility. I would recommend introduce something like
> 
> ```cs
> public double MinRatio { get; set; }
> public double MaxRaio { get; set; }
> ```
> 
> for layout item.
> 
> when bounds ratio change, find the first item that fall into it's range. Any thoughts?

Having users study the relationship between aspect ratio values and shapes can degrade the development experience. 
It might be a good idea to apply the relationship between **Items** and **ItemsSource** to your suggestion.

**Comment by WCKYWCKF** [comment]
Features have been added, please help review.

File: src/Ursa/Controls/AspectRatioLayout/AspectRatioLayout.cs [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+{
+    public static readonly StyledProperty<List<AspectRatioLayoutItem>> ItemsProperty =
+        AvaloniaProperty.Register<AspectRatioLayout, List<AspectRatioLayoutItem>>(
+            nameof(Items));
+
+    public static readonly StyledProperty<double> AspectRatioChangeAmbiguityProperty =
+        AvaloniaProperty.Register<AspectRatioLayout, double>(
+            nameof(AspectRatioChangeAmbiguity), 0.2);
+
+    public static readonly StyledProperty<AspectRatioMode> CurrentAspectRatioModeProperty =
```
Comment (rabbitism) [inline src/Ursa/Controls/AspectRatioLayout/AspectRatioLayout.cs]: Should be a readonly DirectProperty.

File: src/Ursa/Controls/AspectRatioLayout/AspectRatioLayout.cs [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+        set => SetValue(ItemsProperty, value);
+    }
+
+    public double AspectRatioChangeAmbiguity
+    {
+        get => GetValue(AspectRatioChangeAmbiguityProperty);
+        set => SetValue(AspectRatioChangeAmbiguityProperty, value);
+    }
+
+    private void UpdataHistory(bool value)
```
Comment (rabbitism) [inline src/Ursa/Controls/AspectRatioLayout/AspectRatioLayout.cs]: typo: Updata -> Update

File: src/Ursa/Controls/AspectRatioLayout/AspectRatioLayoutItem.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    public double EndAspectRatioValue
+    {
+        get => GetValue(EndAspectRatioValueProperty);
+        set => SetValue(EndAspectRatioValueProperty, value);
+    }
+
+
+    private bool _isUseAspectRatioRange;
+
+    public static readonly DirectProperty<AspectRatioLayoutItem, bool> IsUseAspectRatioRangeProperty =
```
Comment (rabbitism) [inline src/Ursa/Controls/AspectRatioLayout/AspectRatioLayoutItem.cs]: An internal CLR property is enough?

**Review by rabbitism — CHANGES_REQUESTED** [review]
Functionally works well, please check these minor issues.

**Comment by WCKYWCKF** [comment]
Four minor issues have been fixed.

**Review by rabbitism — APPROVED** [review]

**Pull request merged**